src/js/modules/articles/selectors.js

```diff
@@ -79,6 +90,52 @@ export const getArticlesByContributor = createSelector(
 );
 
 /**
+ * The selector returns an object that contains all articles an illustrator has
+ *   illustrated for (featured media only).
+ */
+export const getIllustratorArticles = createSelector(
+  [ getIllustratorIllustrations, getArticlesWithContributors ],
+  (illustrations, articles) => {
+    const articleIds = Object
+      .values(illustrations)
+      .map(illustration => illustration.articleId);
+    return Object.filter(articles, articleObject => {
+      return articleIds.includes(articleObject.id);
+    });
+  }
+);
+
+/**
+ * The selector returns an object that contains all articles a photographer has
+ *   photographed for (featured media only).
+ */
+export const getPhotographerArticles = createSelector(
+  [ getPhotographerPhotographs, getArticlesWithContributors ],
+  (photographs, articles) => {
+    const articleIds = Object
+      .values(photographs)
+      .map(photograph => photograph.articleId);
+    return Object.filter(articles, articleObject => {
+      return articleIds.includes(articleObject.id);
+    });
+  }
+);
+
+/**
+ * The selector returns a media object for the featured media of a requested
+ *   article.
+ */
+export const getArticleFeaturedMedia = createSelector(
+  [ getArticleFromRequestedSlug, getMedia ],
+  (article, media) => {
+    const matchedMedia = Object.filter(media, mediaObject => {
```

Comment:
This is not a great way of searching for stuff. I'd just do `Object.values(media).find(mediaObject => mediaObject.isFeatured && mediaObject.articleId === article.id)`